pkg/bpm/bpm.go

```diff
@@ -76,20 +77,25 @@ type Stdio struct {
 
 // BackgroundProcessManager manages all background processes
 type BackgroundProcessManager struct {
-	deathSig    *sync.Map
-	identifiers *sync.Map
-	stdio       *sync.Map
-
+	deathSig         *sync.Map
+	identifiers      *sync.Map
+	stdio            *sync.Map
 	metricsCollector *metricsCollector
+	log              logr.Logger
 }
 
 // NewBackgroundProcessManager creates a background process manager
 func NewBackgroundProcessManager(registry prometheus.Registerer) BackgroundProcessManager {
+	rootLogger, err := clog.NewDefaultZapLogger()
+	if err != nil {
+		stdlog.Fatal("failed to create root logger", err)
+	}
```

Comment:
please do not create another logger, acquire the logger as a parameter in the constructor.